src/ui/components/shared/UserSettingsModal/PreferencesSettings.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useState } from "react";
+import { connect, ConnectedProps } from "react-redux";
+
+import { updateEnableRepaint } from "protocol/enable-repaint";
+import { handleIntercomLogout } from "ui/utils/intercom";
+import useAuth0 from "ui/utils/useAuth0";
+import hooks from "ui/hooks";
+import * as actions from "ui/actions/app";
+import * as selectors from "ui/reducers/app";
+import { UIState } from "ui/state";
+import { SettingsTabTitle } from "ui/state/app";
+import { ApiKey, UserSettings } from "ui/types";
+
+import APIKeys from "../APIKeys";
+import SettingsModal from "../SettingsModal";
+import { Settings } from "../SettingsModal/types";
+import { SettingsBodyHeader } from "../SettingsModal/SettingsBody";
+
+import { getFeatureFlag } from "ui/utils/launchdarkly";
+import { AvatarImage } from "ui/components/Avatar";
+import { prefs } from "ui/utils/prefs";
+import Checkbox from "../Forms/Checkbox";
+import { disableEmailPreference, EmailPreference, enableEmailPreference } from "ui/hooks/users";
+import Radio from "../Forms/Radio";
+
+const EMAIL_NOTIFICATIONS = {
+  [EmailPreference.COLLABORATOR_REQUESTS]: "When somebody invites you to collaborate on a replay",
+  [EmailPreference.REPLAY_COMMENTS]: "When somebody comments on your replay",
+  [EmailPreference.NEW_TEAM_INVITE]: "When you're invited to a new team",
+};
+
+function Notification({
+  content,
+  preference,
+  emailPreferences,
+}: {
+  content: string;
+  preference: EmailPreference;
+  emailPreferences: EmailPreference[];
+}) {
+  const updateUserEmailPreferences = hooks.useUpdateUserEmailPreferences();
```

Comment:
maybe call these `update`? It makes them a lot easier to read and locally it's the only thing that `update` could possible mean I think.